src/kaczmarz/_variants.py

```diff
@@ -26,10 +26,14 @@ class Cyclic(kaczmarz.Base):
     def __init__(self, *base_args, **base_kwargs):
         super().__init__(*base_args, **base_kwargs)
         self._row_index = -1
+        self._order = range(self._n_rows)
+
+    def _set_order(self, order):
+        self._order = order
```

Comment:
Get rid of `_set_order` and change `__init__` to take an optional argument of `order` with a default value of `None`. Then, replace line 29 with

```python
if order is None:
    order = range(self._n_rows)
self._order = order
```